### PR TITLE
optimize Decilm 7b

### PR DESCRIPTION
**Pull request by qiuxin2012**
## Description

optimize Decilm 7b

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/convert.py [hunk 1]
```diff
@@ -932,4 +932,16 @@ def safe_bmm_fwd(*args, **kwargs):
...
+    elif model.config.model_type == "deci":
+        modeling_module_name = model.__class__.__module__
+        module = importlib.import_module(modeling_module_name)
+        convert_forward(model,
+                        module.LlamaRMSNorm,
+                        llama_rms_norm_forward)
+        convert_forward(model,
+                        module.LlamaMLP,
+                        llama_mlp_forward)
+        convert_forward(model,
```
Comment (qiuxin2012) [inline python/llm/src/bigdl/llm/transformers/convert.py:944]: Logits are different if using this, we can't use llama's attention forward directly.

Comment (qiuxin2012) [reply python/llm/src/bigdl/llm/transformers/convert.py:945]: fixed with decilm_attention_forward_4_35_2

File: python/llm/src/bigdl/llm/transformers/models/decilm.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    if is_decode:
+        attn_output = F.scaled_dot_product_attention(query_states, key_states, value_states,
+                                                     is_causal=False,
+                                                     attn_mask=attention_mask)
+        attn_output = attn_output.contiguous().view(bsz, q_len, self.hidden_size)
+
+    else:
+        attn_output = F.scaled_dot_product_attention(query_states, key_states, value_states,
+                                                     is_causal=attention_mask is None,
+                                                     attn_mask=attention_mask)
```
Comment (rnwang04) [inline python/llm/src/bigdl/llm/transformers/models/decilm.py:159]: if `attention_mask` is not None, there may be error of flash attention on PVC.
However, we can solve this issue later.

**Review by rnwang04 — APPROVED** [review]
In general LGTM.

**Review by sgwhat — APPROVED** [review]
